kernel/src/fs/utils/systree_inode.rs

```diff
@@ -582,31 +593,41 @@ impl<KInode: SysTreeInodeTy + Send + Sync + 'static> Inode for KInode {
 }
 
 // Update AttrDentryIter to filter by min_ino
-struct AttrDentryIter {
-    attrs: Vec<SysAttr>,
+struct AttrDentryIter<'a> {
+    attrs_and_node: Option<(&'a SysAttrSet, &'a dyn SysNode)>,
```

Comment:
Actually `&'a SysAttrSet` should be replaced with an iterator? So we don't need to call `nth` to get an item?